lib/utils.js

```diff
@@ -19,6 +19,35 @@ function shallowCopy(src, dest) {
   return dest;
 }
 
+function take(source, fromRight) {
+  return fromRight ? source.pop() : source.shift();
+}
+
+/**
+ * Takes a list of arguments and maps them to a list of names
+ * in an alternating either a left-to-right, or right-to-left direction.
+ */
+function resolveArgs(argumentsObject, nameMap, rightToLeft) {
+  var result = {};
+
+  var takeFromRight = rightToLeft;
+  var args = Array.prototype.slice.call(argumentsObject);
+
+  while (nameMap.length) {
+    var value = null;
+    var name = take(nameMap, takeFromRight);
+
+    if (args.length > 0) {
+      value = take(args, takeFromRight);
+      takeFromRight = !takeFromRight;
```

Comment:
@robertjd What errors were you seeing? I'm actually seeing errors right now when using both the SDK master and express master. Are you seeing the same?